pkgs/applications/networking/p2p/stig/default.nix

```diff
@@ -0,0 +1,49 @@
+{ lib
+, fetchFromGitHub
+, python
+}:
+
+with python.pkgs; buildPythonApplication rec {
+  pname = "stig";
+  version = "0.10.1a";
+
+  src = fetchFromGitHub {
+    owner = "rndusr";
+    repo = "stig";
+    rev = "v${version}";
+    sha256 = "076rlial6h1nhwdxf1mx5nf2zld5ci43cadj9wf8xms7zn8s6c8v";
+  };
+
+  postPatch = ''
+    substituteInPlace setup.py \
+      --replace "urwidtrees>=1.0.3dev0" "urwidtrees"
+  '';
+
+  propagatedBuildInputs = [
```

Comment:
TBH I'm not sure I totally understand the difference between a propagated build input and a normal build input. Does this have the effect over the `PYTHONPATH` env var when a package is used as a dependency?

Reply:
I think it does. `propagatedBuildInputs`  should be equivalent to `buildInput`, with the only difference being that `propagated` also adds the packages as a dependency to all reverse-dependencies. Not entirely sure if there is other special casing, you'd have to try. Usually you use `propagated` for python libraries since the dependencies need to be propagated all the way up to the final python environment. The same is not the case for applications.